### PR TITLE
Refactored: Changed Snap methods from "namespace with anonymous" func…

### PR DESCRIPTION
**Pull request by grypho**
…tions to "const class with default method" to allow declaration merging that is used in Snap Plugins. (All code still works the same way, test needn't be changed)

Please fill in this template.

- [ X] Use a meaningful title for the pull request. Include the name of the package modified.
- [ X] Test the change in your own code. (Compile and run.)
- [ X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/snapsvg/index.d.ts [hunk 1]
```diff
@@ -1,74 +1,109 @@
...
-    export function parsePathString(pathString:string):Array<any>;
-    export function parsePathString(pathString:Array<string>):Array<any>;
-    export function parseTransformString(TString:string):Array<any>;
-    export function parseTransformString(TString:Array<string>):Array<any>;
-
-    export function closest(x: number, y: number, X: number, Y: number): boolean;
-
-    export interface RGB {
+
+declare module "snapsvg" {
```
Comment (DanielRosenwasser) [inline types/snapsvg/index.d.ts]: I don't think you should be using an ambient module declaration. I think `export as namespace` is indeed the more appropriate form here.

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/snapsvg/index.d.ts (hunk 1 above)
Comment (grypho) [reply types/snapsvg/index.d.ts]: Hi Daniel, I agree with you that a namespace looks cleaner than an ambient declaration. But the current TypeScript version does not support Declaration Merging when a namespace has the same name as a function out of the namespace.

Snap, by architecture, allows plugins that add certain functions to elements. My usecase is to add "translate", "rotate" and "scale" functions to the paper interface in JS. When taking that plugin scheme to TS, Declaration Merging is the appropriate way:

```namespace Snap {
    export interface Paper {
        translate(dx: Number, dy: Number): Snap.Paper;
        scale(sx: Number, sy: Number): Snap.Paper;
        rotate(angle: Number, cx: Number, cy: Number): Snap.Paper;
    }
    export interface Element {
        translate(dx: Number, dy: Number): Snap.Element;
        scale(sx: Number, sy: Number): Snap.Element;
        rotate(angle: Number, cx: Number, cy: Number): Snap.Element;
    }
}```

As long as the Snap(....) functions exist, overloading of both interfaces is not possible. That's the reason why /jquery/index.ts uses the same mechanics as an ambient module.

If you can provide a way, how Declaration Merging works with the current TSC version, I would be happy to use it and to cancel my pull request.

Comment (DanielRosenwasser) [reply types/snapsvg/index.d.ts]: Users can use module augmentation to reopen the interfaces:

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has merge conflicts and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Pull request closed without merging**